### PR TITLE
[React] Upgrade react-jhipster 0.15.0

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #14390 

cc @pascalgrimaud @gzsombor @deepu105

[cause](https://github.com/jhipster/react-jhipster/commit/1266c1e27dc35e355b5811d6a9506aa37c81bfe3)
---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/entity-client/templates/react/src/main/webapp/app/entities/entity.tsx.ejs [hunk 1]
```diff
@@ -72,7 +72,7 @@ export const <%= entityReactName %> = (props: I<%= entityReactName %>Props) => {
   const [search, setSearch] = useState('');
     <%_ } _%>
     <%_ if (pagination !== 'no') { _%>
-  const [paginationState, setPaginationState] = useState(overridePaginationStateWithQueryParams(getSortState(props.location, ITEMS_PER_PAGE), props.location.search));
+  const [paginationState, setPaginationState] = useState(overridePaginationStateWithQueryParams(getSortState(props.location, ITEMS_PER_PAGE, null), props.location.search));
```
Comment (mshima) [inline generators/entity-client/templates/react/src/main/webapp/app/entities/entity.tsx.ejs:75]:
```suggestion
  const [paginationState, setPaginationState] = useState(overridePaginationStateWithQueryParams(getSortState(props.location, ITEMS_PER_PAGE, '<%= primaryKey.name %>'), props.location.search));
```

File: generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs [hunk 2]
```diff
@@ -38,7 +38,7 @@ import { IRootState } from 'app/shared/reducers';
 export interface IUserManagementProps extends StateProps, DispatchProps, RouteComponentProps<any> {}
 
 export const UserManagement = (props: IUserManagementProps) => {
-  const [pagination, setPagination] = useState(overridePaginationStateWithQueryParams(getSortState(props.location, ITEMS_PER_PAGE), props.location.search));
+  const [pagination, setPagination] = useState(overridePaginationStateWithQueryParams(getSortState(props.location, ITEMS_PER_PAGE, null), props.location.search));
```
Comment (mshima) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs:41]:
```suggestion
  const [pagination, setPagination] = useState(overridePaginationStateWithQueryParams(getSortState(props.location, ITEMS_PER_PAGE, '<%= primaryKey.name %>'), props.location.search));
```

**Review by mshima — COMMENTED** [review]
Custom id should be passed?

File: generators/entity-client/templates/react/src/main/webapp/app/entities/entity.tsx.ejs (hunk 1 above)
Comment (qmonmert) [reply generators/entity-client/templates/react/src/main/webapp/app/entities/entity.tsx.ejs:75]: ok

File: generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs (hunk 2 above)
Comment (qmonmert) [reply generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs:41]: ok

**Comment by qmonmert** [comment]
@mshima your suggests don't work 

image

**Comment by mshima** [comment]
@qmonmert my bad. `user-management.tsx.ejs` should be `user.primaryKey.name` instead.

**Comment by pascalgrimaud** [comment]
Adding a bounty as the fix was needed quickly for the release v7
Thanks and don't forget to claim it @qmonmert

**Comment by qmonmert** [comment]
thanks @pascalgrimaud ;)

**Comment by qmonmert** [comment]
@pascalgrimaud https://opencollective.com/generator-jhipster/expenses/36494 :)

**Comment by pascalgrimaud** [comment]
@qmonmert : approved ! thanks again for the help!

**Pull request merged**